### DISCUSSION TITLE
Internal HTTP adress of Airflow in Standard Docker-Compose

### DISCUSSION THREAD

**Discussion by PatrickvanZalm** [apache/airflow#22110] [Q&A]
Hi all,

For our Academic Lab we are interested in using airflow. Im using the docker-compose.yaml as [provided on the airflow website](https://airflow.apache.org/docs/apache-airflow/2.2.4/docker-compose.yaml).

I have developed a Python script that will use the API to trigger a DAG. This works using localhost:8080. Now I have put that Python script in a Docker Container, which I did add to the Airflow Docker-compose.yaml file. Here is where I am lost:

I do know that I should not use localhost:8080 here since my container and the airflow container(s) run in the same Docker-network, but I am completly lost in the sense that I do not know where to send the http request to instead.

For clarification: the API Trigger request using localhost looks like this:
`http://localhost:8080/api/v1/dags/MyDAG/dagRuns`

Thank you in advance,
Patrick

**Comment by potiuk** [comment]
Docker compose by definition creates host names that are equal to names of services (so `airflow') but in newer docker-compose you can use "hostname" apparently https://stackoverflow.com/questions/29924843/how-do-i-set-hostname-in-docker-compose

**Reply by PatrickvanZalm** [reply]
Hi Potiuk,

first of all thank you very your very prompt reply.

I tried the following four http requests according to your reply, but none did trigger the DAG (while localhost does).
`curl -X GET --user "airflow:airflow" "http://hostname:8080/api/v1/dags"`
`curl -X GET --user "airflow:airflow" "hostname:8080/api/v1/dags"`
`curl -X GET --user "airflow:airflow" "http://airflow:8080/api/v1/dags"`
`curl -X GET --user "airflow:airflow" "airflow:8080/api/v1/dags"`

Thus I wonder; should I also open up something for my own added docker? In the compose it would look like the following:
`services:
  copy-data-bot:
    build: ./product
    volumes:
      - /mnt/MyDrive/:/SharedDrive/
    depends_on:
      <<: *airflow-common-depends-on
      airflow-init:
        condition: service_completed_successfully`

**Reply by PatrickvanZalm** [reply]
Screenshot for easier reading...

![image](https://user-images.githubusercontent.com/56233030/157440406-3cb7a6b7-e014-4dc7-9c74-d8e575dabcb1.png)

**Reply by potiuk** [reply]
Unfortunattely can't help more. The problem with docker compose is that you are very much on your own and you need to debug it yourself - because every docker-compose is different. That's why our quick-start is just this - quick-start and when you try to make anything more serious with it. Generally as community we recommend not to do so (and there is no good "support" for the compose beynd the quick-start).

We even put a warning https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#production-readiness about it.
Maybe someone will be able to help here, but docker-compose has a lot of intricacies and networking issues and it's extremely bad in getting someon help on your iteration over the docker compose. I do not know docker-compose extremely well (but good enough) but honestly I am lost 90% of time when someone ask a question about some networking not working, host mapping etc.

Sorry for not being to helpful but every time disusssion gets to the point my advice is to either deeply learn about the networking and details of compose on your own - to understand the ports/networking, or switch to K8S and our Helm Chart - https://airflow.apache.org/docs/helm-chart/stable/index.html - which is much better supported and behaves in a predictable way (though of course it requires quite a bit learning too).

**Reply by PatrickvanZalm** [reply]
Alright, that is some solid advice too. Guess spending time on docker-compose troubleshooting is better spent as you seem to suggest.

Again, thank you for your time.
